packages/amplify_storage_s3/ios/Classes/FlutterListRequest.swift

```diff
@@ -15,16 +15,21 @@
 
 import Foundation
 import Amplify
+import amplify_core
 
 struct FlutterListRequest {
     var options: StorageListRequest.Options?
     init(request: Dictionary<String, AnyObject>) {
         self.options = setOptions(request: request)
     }
-    
-    static func isValid(request: Dictionary<String, AnyObject>) -> Bool {
-        let valid: Bool = request["path"] is String?
-        return valid
+
+    static func validate(request: Dictionary<String, AnyObject>) throws {
+        let validationErrorMessage = "List request malformed."
+        if !(request["path"] is String?) {
```

Comment:
probably no `?` here

Reply:
I think this should be checking 'if path is not null, it must be a String'.